contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js

```diff
@@ -2,8 +2,43 @@ import difference from 'lodash/difference';
 import union from 'lodash/union';
 import { NOVALUE } from 'shared/constants';
 import client from 'shared/client';
-import { MOVE_POSITIONS } from 'shared/data/constants';
+import { RELATIVE_TREE_POSITIONS } from 'shared/data/constants';
 import { ContentNode, Tree } from 'shared/data/resources';
+import { promiseChunk } from 'shared/utils';
+
+/**
+ * @param context
+ * @param {String} channelId
+ * @return {Promise<Object>}
+ */
+function getChannel(context, channelId) {
+  return new Promise((resolve, reject) => {
+    const channel = context.rootGetters['channel/getChannel'](channelId);
+
+    if (channel) {
+      resolve(channel);
+    } else {
+      reject();
+    }
+  });
+}
+
+/**
+ * @param {String} id
+ * @return {Promise<Object[]>}
+ */
+function loadParentRecursively(id) {
```

Comment:
Could we tie this in with the loadAncestors more? They seem like they aim to achieve the same thing

Reply:
The `loadAncestors` action is using this! If I recall, we had a discussion about this. I've removed `rght` from the frontend. I'm just realizing now that perhaps I should announce that on #studio-refactors

Reply:
Yep! Would it make sense to just call loadParentRecursively and remove the loadAncestors function completely?

Reply:
I see what you're saying. This is just a helper for `loadAncestors`, it's not exposed as an action. But I think I can consolidate them. I think I consolidated them similarly for the getter. I'll take a look.